Objects/abstract.c

```diff
@@ -819,6 +819,21 @@ binary_op(PyObject *v, PyObject *w, const int op_slot, const char *op_name)
     PyObject *result = binary_op1(v, w, op_slot);
     if (result == Py_NotImplemented) {
         Py_DECREF(result);
+
+        if (strcmp(op_name, ">>") == 0 && \
```

Comment:
I need to re-read the whole file myself to see if there may be a better place to inject this check, but in the mean time, at least compare against the numeric `op_slot` here, rather than the string value.